### PR TITLE
Enable Guardian Weekly guest checkout

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
This PR removes the requirement to sign in before purchasing a GW sub or gift sub

[**Trello Card**](https://trello.com/c/YB2VSB5h/3880-implement-gw-gifting-and-non-gifting-guest-checkout-not-an-ab-test)

## Why are you doing this?
- To remove friction from the checkout process
- Changes to the Identity sign up process mean that users can no longer be logged in until they verify their email. This would make the old flow completely impractical
- It allows customer service representatives (CSRs) to use these checkouts to purchase subscriptions for customers rather than having to use the subscribe site

## Is this an AB test?
- [ ] Yes
- [x] No

### CODE REVIEW

File: support-frontend/app/controllers/WeeklySubscriptionForm.scala [hunk 1]
```diff
@@ -39,28 +40,31 @@ class WeeklySubscriptionForm(
...
   implicit val a: AssetsResolver = assets
 
-  def displayForm(orderIsAGift: Boolean): Action[AnyContent] = authenticatedAction(subscriptionsClientId).async { implicit request =>
+  def displayForm(orderIsAGift: Boolean): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
     implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    identityService.getUser(request.user.minimalUser).fold(
+    val maybeIdUser: EitherT[Future, String, Option[IdUser]] = request.user match {
+      case Some(user) =>
+        identityService.getUser(user.minimalUser).map(Some(_))
+      case _ => EitherT.rightT(None)
```
Comment (johnduffell) [inline support-frontend/app/controllers/WeeklySubscriptionForm.scala:48]:
```suggestion
      case None => EitherT.rightT(None)
```

File: support-frontend/assets/helpers/subscriptionsForms/guestCheckout.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 // @flow
 
-import type { CheckoutState } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
+import type { CheckoutState, WithDeliveryCheckoutState } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
 import { getUserTypeFromIdentity } from 'helpers/identityApis';
 import type { UserTypeFromIdentityResponse } from 'helpers/identityApis';
 import { setUserTypeFromIdentityResponse } from 'helpers/subscriptionsForms/formActions';
 
 export const checkIfEmailHasPassword = (email: string) =>
```
Comment (johnduffell) [inline support-frontend/assets/helpers/subscriptionsForms/guestCheckout.js:8]: I realise it's existing, but it's not clear what this function does as it returns void, it looks more like `eventuallySaveUserTypeInRedux` or something to that effect?

**Review by johnduffell — APPROVED** [review]
looks good, just a couple of comments for your perusal 👍

File: support-frontend/assets/helpers/subscriptionsForms/guestCheckout.js (hunk 2 above)
Comment (rupertbates) [reply support-frontend/assets/helpers/subscriptionsForms/guestCheckout.js:8]: Yes it is a bit confusing, I'll have a look at refactoring

**Pull request merged**